bootstrap/v2/pkg/apis/kferrors.go

```diff
@@ -39,3 +40,25 @@ func (e *KfError) Error() string {
 	return fmt.Sprintf(" (kubeflow.error): Code %d with message: %v",
 		e.Code, e.Message)
 }
+
+// NewKfErrorWithMessage will propogate the error with the given message.
+//
+// TODO(jlewi): Not sure this is the best way to propogate the error messages and turn them
+// into KfErrors. There was a lot of code that was doing this but not asserting that the error
+// was a KfError which was causing segmentation faults so I wrote this helper method.
+func NewKfErrorWithMessage(e error, msg string) error {
+	kErr, ok := e.(*KfError)
+
+	if !ok {
+		log.Infof("Error is not a KfError; %v", e)
+
+		return &KfError{
+			Code:    int(UNKNOWN),
+			Message: msg + "; " + e.Error(),
```

Comment:
Final thought: Do you think that maybe we should use errors.Wrap here?
Or we don't want to include a stacktrace ?

Reply:
I think we need to rethink how we handle errors and be more consistent and thorough and ideally include stacktrace info.

Including stacktraces is definitely a good idea.

This is mostly a quick fix to deal with the fact that we have a bunch of unsafe code right now.